fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/metadata/expressions/KeyExpression.java

```diff
@@ -119,13 +119,23 @@
     List<Descriptors.FieldDescriptor> validate(@Nonnull Descriptors.Descriptor descriptor);
 
     /**
-     * Returns the number of items in each KeyValue that will be returned.
-     * When we support concatenate, this will the count of non-nested lists, i.e.
-     * this will be value of evaluate(r).get(i).toList().size() for any i or r.
+     * Returns the number of items in each KeyValue that will be returned. For key expressions that support
+     * {@link FanType#Concatenate}, this will the count of non-nested lists, i.e. this will be value of
+     * evaluate(r).get(i).toList().size() for any i or r.
+     *
      * @return the number of elements that will be produced for every key
      */
     int getColumnSize();
 
+    /**
+     * Returns whether or not the number of columns produced by the evaluation of a key expression can
+     * vary based upon data in the record itself.  Specifically, this will return {@code false} for
+     * expressions involving {@link FanType#Concatenate}.
+     *
+     * @return whether or not the column count is fixed or may vary based upon the data in the record
+     */
+    boolean isFixedColumnSize();
```

Comment:
Maybe I don't understand this method, but aren't the number of columns (as the term is used in `getColumnSize`) fixed for `FanType.Concatenate` as well? In particular, I think it produces one (larger than usual) column that combines all of the elements of the repeated field.

Reply:
Right. The primary key will have a nested tuple for that column. So perhaps I was mistaken and there aren't any cases where the primary key isn't a tuple of the same length as `getColumnSize`.

Reply:
Ok. I can remove this.